dom.go

```diff
@@ -454,6 +489,24 @@ func (tl *TokenList) Set(s []string) {
 	tl.SetString(strings.Join(s, " "))
 }
 
+type attribute struct {
+	key   string
+	value string
+}
+
+func (a attribute) Value() string {
+	return a.value
+}
+
+func (a attribute) Key() string {
+	return a.key
+}
+
+type Attribute interface {
```

Comment:
Let's use a map. A slice suggests order. The fact that changes to the map won't update the DOM isn't an issue.